patroni/postgresql.py

```diff
@@ -57,15 +58,17 @@ def __init__(self, config):
         self.pg_rewind = config.get('pg_rewind') or {}
         self.callback = config.get('callbacks') or {}
         self.use_slots = config.get('use_slots', True)
-        self.schedule_load_slots = self.use_slots
-        self.recovery_conf = os.path.join(self.data_dir, 'recovery.conf')
-        self.configuration_to_save = (os.path.join(self.data_dir, 'pg_hba.conf'),
-                                      os.path.join(self.data_dir, 'postgresql.conf'))
-        self.postmaster_pid = os.path.join(self.data_dir, 'postmaster.pid')
-        self.trigger_file = config.get('recovery_conf', {}).get('trigger_file') or 'promote'
-        self.trigger_file = os.path.abspath(os.path.join(self.data_dir, self.trigger_file))
-
-        self._pg_ctl = ['pg_ctl', '-w', '-D', self.data_dir]
+        self._schedule_load_slots = self.use_slots
+        self._postgresql_conf = os.path.join(self._data_dir, 'postgresql.conf')
+        self._postgresql_base_conf_name = 'postgresql-base.conf'
+        self._postgresql_base_conf = os.path.join(self._data_dir, self._postgresql_base_conf_name)
+        self._recovery_conf = os.path.join(self._data_dir, 'recovery.conf')
+        self._configuration_to_save = (self._postgresql_conf, os.path.join(self._data_dir, 'pg_hba.conf'))
```

Comment:
If we don't include postgresql_base_conf here, than we'll break WAL-E restore from the backup (it will backup postgesql.conf, which would include postgresql-base.conf, but not postgresql-base.conf, since it excludes *.conf from the PGDATA, resulting in copying the wrong postgresql.conf to postgresql-base.conf).